src/comestible_inv_listitem.cpp

```diff
@@ -0,0 +1,771 @@
+            };
+        case COLUMN_SORTBY_CATEGORY:
+            return []( const itm * d1, const itm * d2, bool & retval ) {
+                assert( d1->cat != nullptr );
+                assert( d2->cat != nullptr );
+                if( d1->cat != d2->cat ) {
+                    retval = *d1->cat < *d2->cat;
+                    return true;
+                } else if( d1->is_category_header() ) {
+                    retval = true;
+                    return true;
+                } else if( d2->is_category_header() ) {
+                    retval = false;
+                    return true;
+                }
+                retval = false;
+                return false;
+            };
+        default:
+            return nullptr;
+    }
+}
+
+comestible_inv_listitem::comestible_inv_listitem( const std::list<item *> &list, int index,
+        comestible_inv_area *area, bool from_vehicle ) :
+    idx( index ),
+    area( area ),
+    id( list.front()->typeId() ),
+    items( list ),
+    name( list.front()->tname( list.size() ) ),
+    name_without_prefix( list.front()->tname( 1, false ) ),
+    autopickup( get_auto_pickup().has_rule( list.front() ) ),
+    stacks( list.size() ),
+    volume( list.front()->volume() * stacks ),
+    weight( list.front()->weight() * stacks ),
+    cat( &list.front()->get_category() ),
+    from_vehicle( from_vehicle )
+{
+    init( g->u.get_consumable_from( *list.front() ) );
+}
+
+void comestible_inv_listitem::init( const item &it )
+{
+    menu_color = it.color_in_inventory();
+    menu_color_dark = c_dark_gray;
+    assert( stacks >= 1 );
+}
+
+comestible_inv_listitem::comestible_inv_listitem( const item_category *cat )
+    : idx()
+    , area()
+    , id( "null" )
+    , autopickup()
+    , stacks()
+    , cat( cat )
+{
+    if( cat != nullptr ) {
+        name = cat->name();
+    }
+    menu_color_dark = c_dark_gray;
+}
+
+bool comestible_inv_listitem::is_category_header() const
+{
+    return items.empty() && cat != nullptr;
+}
+
+bool comestible_inv_listitem::is_item_entry() const
+{
+    return !items.empty();
+}
+
+
+/*******    FOOD    *******/
+
+
+comestible_inv_listitem_food::comestible_inv_listitem_food( const item_category *cat ):
+    comestible_inv_listitem( cat )
+{
+}
+
+comestible_inv_listitem_food::comestible_inv_listitem_food( const std::list<item *> &list,
+        int index,
+        comestible_inv_area *area, bool from_vehicle ): comestible_inv_listitem( list, index, area,
+                    from_vehicle )
+{
+    init( g->u.get_consumable_from( *list.front() ) );
+}
+
+void comestible_inv_listitem_food::init( const item &it )
+{
+    player &p = g->u;
+    islot_comestible edible = get_edible_comestible( p, it );
+
+    menu_color = it.color_in_inventory();
+
+    // statuses
+    const auto &comest = it.get_comestible();
+    cond.reserve( cond_size );
+    const bool eat_verb = it.has_flag( "USE_EAT_VERB" );
+    const bool is_food = eat_verb || comest->comesttype == "FOOD";
```

Comment:
I'm getting a crash here on opening the menu, I'm assuming comest is uninitialized.

Reply:
Do you know what item it crashes on? Seems to work with most things.
This edible vs comestible really needs to be documented =\